File: CFLServerData/CFL_Data/Defunts/Defunt.cs

using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using CFL_1.CFL_Data.Etat_civil;

namespace CFL_1.CFL_Data.Defunts
{
    public class Defunt : Personne
    {
        public Deces Deces { get; set; } = new Deces();

        public Naissance Naissance { get; set; } = new Naissance();

        public Filiation Filiation { get; set; } = new Filiation();

        [Display(Name = "Situation familiale")]
        public SituationFamiliale SituationFamiliale { get; set; } = new SituationFamiliale();
        
        public Pouvoir Pouvoir { get; set; } = new Pouvoir();

        public List<OperationFune> OperationsFuneraires { get; set; } = new List<OperationFune>();
    }
}

File: src/cflserver/cfl_data/defunt/defunt.py
from mongoengine import *

import src.cflserver.cfl_data.etat_civil.personne as personne
import src.cflserver.cfl_data.defunt.operation_fune as opfune
import src.cflserver.cfl_data.defunt.deces as deces

class Defunt(Document):

    personne = ReferenceField(personne.Personne, default=None)
    deces = ReferenceField(deces.Deces, default=None)
    situation = StringField(default="") # enumSituation

    operations = ListField(opfune.OperationFune, default=None)
